### PR TITLE
refactor: add env creation on yarn dx

### LINKED ISSUE #8591: Create a full independent docker environment to run the project on
### Is your proposal related to a problem?

I tried to get one "good first issue" and was kind of annoying to setup a database externally just to start to contribute to the project

### Describe the solution you'd like

It would be nice to have a `docker-compose.yml` setup with everything that we need to run the app, something that with a `docker-compose up` on the repo's source we'll be able to run the project easily.

### PR DESCRIPTION
**Pull request by wesleymatosdev**
- build: add script to create new .env files
- build: add env script creation on turbo dx

## What does this PR do?

Fixes #8591

/claim #8591

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)
- This change requires a documentation update

## How should this be tested?

- [ ] **[CASE-0]** Testing that it'll not affect what is already running
  - Get your repo that you already spin up (you already have the configs and don't need to create any envs/seed anything)
  - Run the app as you would do normally
  - These changes should not affect what already works.
- [ ] **[CASE-1]** Testing in a brand-new repo
  - Clone the repo
  - Install the dependencies using `yarn`
  - Run the app using `yarn dx`
- [ ] **[CASE-2]** Testing in a repo that already exists
  - Run `git clean -dX -f` (this will delete everything on your repo that git does not track)
  - Install the dependencies using `yarn`
  - Run the app using `yarn dx`

## Checklist

- I haven't checked if my PR needs changes to the documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "dx": "turbo run dx",
     "i-dev": "infisical run -- turbo run dev --scope=\"@calcom/web\"",
     "i-dx": "infisical run -- turbo run dx",
+    "init-env": "[ ! -f .env ] && cp .env.example .env && echo NEXTAUTH_SECRET='\"'$(openssl rand -base64 32)'\"' >> .env && echo CALENDSO_ENCRYPTION_KEY='\"'$(openssl rand -base64 24)'\"' >> .env && [ ! -f .env.appStore ] && cp .env.appStore.example .env.appStore",
```
Comment (ologbonowiwi) [inline package.json:43]:
```suggestion
    "init-env": "[ ! -f .env ] && cp .env.example .env && echo NEXTAUTH_SECRET='\"'$(openssl rand -base64 32)'\"' >> .env && echo CALENDSO_ENCRYPTION_KEY='\"'$(openssl rand -base64 24)'\"' >> .env,
    "init-app-store-env": "[ ! -f .env.appStore ] && cp .env.appStore.example .env.appStore",
```

File: turbo.json [hunk 2]
```diff
@@ -159,6 +160,10 @@
       ],
       "outputs": ["dist/**", "build/**"]
     },
+    "//#init-env": {
+      "cache": false,
+      "outputs": ["./.env", "./.env.appStore"]
+    },
```
Comment (ologbonowiwi) [inline turbo.json:166]:
```suggestion
    "//#init-env": {
      "cache": false,
      "outputs": ["./.env"]
    },
    "//#init-app-store-env": {
      "cache": false,
      "outputs": ["./.env.appStore"]
    }
```

File: turbo.json [hunk 3]
```diff
@@ -93,6 +93,7 @@
       "cache": false
     },
     "dx": {
+      "dependsOn": ["//#init-env"],
```
Comment (ologbonowiwi) [inline turbo.json:96]:
```suggestion
      "dependsOn": ["//#init-env", "//#init-app-store-env"],
```

File: package.json [hunk 4]
```diff
@@ -40,6 +40,8 @@
     "dx": "turbo run dx",
     "i-dev": "infisical run -- turbo run dev --scope=\"@calcom/web\"",
     "i-dx": "infisical run -- turbo run dx",
+    "init-env": "[ ! -f .env ] && cp .env.example .env && echo NEXTAUTH_SECRET='\"'$(openssl rand -base64 32)'\"' >> .env && echo CALENDSO_ENCRYPTION_KEY='\"'$(openssl rand -base64 24)'\"' >> .env,
```
Comment (ologbonowiwi) [inline package.json:43]:
```suggestion
    "init-env": "[ ! -f .env ] && cp .env.example .env && echo NEXTAUTH_SECRET='\"'$(openssl rand -base64 32)'\"' >> .env && echo CALENDSO_ENCRYPTION_KEY='\"'$(openssl rand -base64 24)'\"' >> .env",
```

**Comment by ologbonowiwi** [comment]
@krumware mentioning you here since we've talked about #8591

**Comment by nicktrn** [comment]
It's a good idea but there's a couple of issues with this.

1. Will produce non-0 exit codes on repeated runs
2. Appends instead of overwrites the vars already present

Some ideas:

1. Maybe turbo doesn't care, but you won't be able to easily chain this, e.g. `yarn run init-env && echo only prints on first run`. Can be fixed by wrapping in if.
2. This WILL confuse people. Values set further up (where the variables first appear) are overridden by the appended ones. Can be fixed by using sed.

But this gets _very_ messy in json, even more so because of yarn v3 run:
```json
{
  "dx:env": "yarn run init-env && yarn run init-app-store-env",
  "init-env": "sh -c 'if [ ! -f .env ]; then cp .env.example .env; sed -i \"s~\\(NEXTAUTH_SECRET=\\).*~\\1$(openssl rand -base64 32)~\" .env; sed -i \"s~\\(CALENDSO_ENCRYPTION_KEY=\\).*~\\1$(openssl rand -base64 24)~\" .env; fi'",
  "init-app-store-env": "sh -c 'if [ ! -f .env.appStore ]; then cp .env.appStore.example .env.appStore; fi'"
}
```

Much easier to read in a separate init script:
```sh
#!/bin/sh
set -e

if [ ! -f .env ]; then
  cp .env.example .env
  # note: this likely complains on macos, may have to use -i"" or -i ""
  sed -i "s~\(NEXTAUTH_SECRET=\).*~\1$(openssl rand -base64 32)~" .env
  sed -i "s~\(CALENDSO_ENCRYPTION_KEY=\).*~\1$(openssl rand -base64 24)~" .env
fi

if [ ! -f .env.appStore ]; then
  cp .env.appStore.example .env.appStore
fi
```
```json
{
  "dx:env": "./init-env.sh"
}
```

**Pull request closed without merging**